src/components/clusterConfiguration/ClusterConfiguration.tsx

```diff
@@ -173,8 +174,14 @@ const ClusterConfiguration: React.FC<ClusterConfigurationProps> = ({ cluster })
 
     // update the cluster validation
     try {
-      const { data } = await patchCluster(cluster.id, values);
+      const params = { ...values };
+      if (!params.pullSecret) {
+        // avoid update to empty value by backend
+        delete params.pullSecret;
+      }
```

Comment:
Even if we reset the field value when hiding the textarea, I'd prefer setting the pull secret only when `isPullSecretEdit` is enabled regardless of the actual value of the pullSecret. This would eventually allow reseting the value with empty string.
```suggestion
      const params = isPullSecretEdit ? _.omit(values, ['pullSecret']) : values;
```

Reply:
vice versa, but ok